esmvalcore/preprocessor/_mask.py

```diff
@@ -377,7 +377,7 @@ def count_spells(data, threshold, axis, spell_length):
         axis += data.ndim
     # Threshold the data to find the 'significant' points.
     if not threshold:
-        data_hits = data
```

Comment:
This line lead to the buggy behavior: Since `data_hits` is interpreted as `bool` later in the code a `0` in `data` leads to (undesirable) `False` in the data and thus to wrong masking.